domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/casetype/mapper/FieldTypeListItemMapper.java

```diff
@@ -4,26 +4,43 @@
 import uk.gov.hmcts.ccd.definition.store.repository.model.FixedListItem;
 
 import java.util.ArrayList;
+import java.util.Comparator;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class FieldTypeListItemMapper {
     public List<FixedListItem> entityToModel(final List<FieldTypeListItemEntity>
-                                                                    fieldTypeListItemEntities) {
+                                                 fieldTypeListItemEntities) {
         List<FixedListItem> fixedListItems = new ArrayList<>();
         fieldTypeListItemEntities.stream().forEach(fieldTypeListItemEntity ->
-                                                   {
-                                                       FixedListItem fixedListItem = new FixedListItem();
-                                                       fixedListItem
-                                                           .setCode(fieldTypeListItemEntity.getValue());
-                                                       fixedListItem
-                                                           .setLabel(fieldTypeListItemEntity.getLabel());
-                                                       fixedListItem
-                                                           .setOrder(fieldTypeListItemEntity.getOrder());
-                                                       fixedListItems.add(fixedListItem);
-                                                   }
-                                                  );
+            {
+                FixedListItem fixedListItem = new FixedListItem();
+                fixedListItem
+                    .setCode(fieldTypeListItemEntity.getValue());
+                fixedListItem
+                    .setLabel(fieldTypeListItemEntity.getLabel());
+                fixedListItem
+                    .setOrder(fieldTypeListItemEntity.getOrder());
+                fixedListItems.add(fixedListItem);
+            }
+        );
+        if (hasADisplayOrder(fixedListItems)) {
+            return applyDisplayOrder(fixedListItems);
```

Comment:
can't we just use a SortedSet like in https://stackoverflow.com/questions/38066291/how-to-define-comparator-on-sortedset-like-treeset